tests/test_trigger_inout.cpp

```diff
@@ -0,0 +1,20 @@
+#include "sm_tests.h"
+
+
+/** @brief Fills [SWS_SM_00020]
+ *         Fills [SWS_SM_00021] */
+TEST_F(smTests, testInternalStateInfluenceAndPropagation) {
+    mySM.triggerInOut.SetSMState(ara::sm::SMStateType::On);
+
+    // SM part
+    EXPECT_TRUE(mySM.triggerInOut.IsTrigger());
+    mySM.triggerInOut.SetTransitionResult(ara::sm::ErrorType::kSuccess, ara::sm::SMStateType::On);
+    mySM.triggerInOut.DiscardTrigger();
```

Comment:
SM part? do you mean that this should be executed from SM?
If so, move it to SM, we will clean SM later

Reply:
Done